File: 0003-数组中重复的数字/Solution2.java
class Solution {
    public int findRepeatNumber(int[] nums) {
        if (nums == null || nums.length == 0) {
            return -1;
        }
        for (int i = 0; i < nums.length; i++) {
            while(nums[i] != i){
                if(nums[i] == nums[nums[i]]){
                    return nums[i];
                }
                int temp = nums[i];
                nums[i]  = nums[temp];
                nums[temp] = temp;
            }
        }
        return -1;
    }
}

File: 0015-三数之和/Solution2.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        ret = []
        used = set()
        for i, num in enumerate(nums):
            if num in used:
                continue
            if num > 0:
                break
            used.add(num)
            result = self.twoSum(nums, i+1, -num)
            for path in result:
                tmp = path[:]
                tmp.append(num)
                ret.append(tmp)
        return ret

    def twoSum(self, nums, index, target):
        helper = {}
        ret = []
        norepeat = set()
        for i in range(index, len(nums)):
            if nums[i] in helper and nums[i] not in norepeat:
                ret.append([nums[i], nums[helper[nums[i]]]])
                norepeat.add(nums[i])
                norepeat.add(nums[helper[nums[i]]])
                continue
            helper[target-nums[i]] = i
        return ret
